code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -51,91 +60,153 @@
+	for(var/mob/living/carbon/C in range(6, owner))
+		if(C != owner)
+			C.adjust_fire_stacks(8)
+			C.IgniteMob()
+	owner.set_fire_stacks(-20)
+	reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
+	return TRUE
+
+/obj/item/clothing/suit/armor/reactive/fire/emp_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>[src] just makes [attack_text] worse, by spewing molten death on [owner]!</span>")
+	playsound(get_turf(owner),'sound/magic/fireball.ogg', 100, TRUE)
+	owner.adjust_fire_stacks(12)
+	owner.IgniteMob()
+	reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
 	return FALSE
 
 //Stealth
 
 /obj/item/clothing/suit/armor/reactive/stealth
 	name = "reactive stealth armor"
 	desc = "An experimental suit of armor that renders the wearer invisible on detection of imminent harm, and creates a decoy that runs away from the owner. You can't fight what you can't see."
+	cooldown_message = "<span class='danger'>The reactive stealth system activates, but is not charged enough to fully cloak!</span>"
+	emp_message = "<span class='warning'>The reactive stealth armor's threat assessment system crashes...</span>"
 
-/obj/item/clothing/suit/armor/reactive/stealth/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
-	if(!active)
-		return FALSE
-	if(prob(hit_reaction_chance))
-		if(world.time < reactivearmor_cooldown)
-			owner.visible_message("<span class='danger'>The reactive stealth system on [owner] activates, but is still recharging its holographic emitters!</span>")
-			return
-		var/mob/living/simple_animal/hostile/illusion/escape/E = new(owner.loc)
-		E.Copy_Parent(owner, 50)
-		E.GiveTarget(owner) //so it starts running right away
-		E.Goto(owner, E.move_to_delay, E.minimum_distance)
-		owner.alpha = 0
-		owner.visible_message("<span class='danger'>[owner] is hit by [attack_text] in the chest!</span>") //We pretend to be hit, since blocking it would stop the message otherwise
-		addtimer(VARSET_CALLBACK(owner, alpha, initial(owner.alpha)), 4 SECONDS)
-		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
-		return TRUE
+/obj/item/clothing/suit/armor/reactive/stealth/cooldown_activation(mob/living/carbon/human/owner)
+	owner.alpha = max(0, owner.alpha - 50)
+	animate(owner, alpha = initial(owner.alpha), time = 3 SECONDS)
+	..()
+
+/obj/item/clothing/suit/armor/reactive/stealth/reactive_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	var/mob/living/simple_animal/hostile/illusion/escape/E = new(owner.loc)
+	E.Copy_Parent(owner, 50)
+	E.GiveTarget(owner) //so it starts running right away
+	E.Goto(owner, E.move_to_delay, E.minimum_distance)
```

Comment:
better variable names

Reply:
i know you didnt write these but this is the best chance we're gonna get for a long time